src/main/java/Duke/activityAndUtility/Event.java

```diff
@@ -0,0 +1,109 @@
+import java.util.Objects;
+
+/**
+ * The {@code Event} class represents an event activity, implementing the {@link Activity} interface. It encapsulates
+ * the status, name, start date and time, and end date and time of an event. This class is designed to manage events,
+ * allowing for them to be printed, named, and marked as complete or incomplete. It ensures that the event's start
+ * date and time are logically before its end date and time.
+ */
+public class Event implements Activity {
+    List<String> act;
+    LocalDate startDate;
+    LocalDate endDate;
+    LocalTime startTime;
+    LocalTime endTime;
+
+    /**
+     * Constructs a new {@code Event} instance with the given status, name, start date and time, and end date and time.
+     * Parses the start and end date,time strings to set the corresponding {@code LocalDate} and {@code LocalTime} fields.
+     * Validates that the start date and time are before the end, throwing a {@code RuntimeException} if not.
+     *
+     * @param status            The initial status of the event (e.g., completed or not).
+     * @param name              The name or description of the event.
+     * @param startDateAndTime  The start date and time string for the event, to be parsed.
+     * @param endDateAndTime    The end date and time string for the event, to be parsed.
+     * @throws RuntimeException if the end date is before the start date.
+     */
+    public Event(String status, String name, String startDateAndTime, String endDateAndTime) {
+        act = new ArrayList<>();
+        act.add(status); // Status
+        act.add(name); // Event name
+        act.add(startDateAndTime);
+        act.add(endDateAndTime);
+
+        LocalDate startDate = DateTimeFormat.getDate(startDateAndTime);
+        LocalTime startTime = DateTimeFormat.getTime(startDateAndTime);
+        LocalDate endDate = DateTimeFormat.getDate(endDateAndTime);
+        LocalTime endTime = DateTimeFormat.getTime(endDateAndTime);
+
+        if (startDate != null && endDate != null) {
+            if (startDate.isAfter(endDate)) {
+                throw new RuntimeException("Finish date ahead of start date");
+            } else if (startDate.isEqual(endDate)) {
+                if (startTime != null && endTime != null && startTime.isAfter(endTime)) {
+                    throw new RuntimeException ("Finish time ahead of start time");
+                }
+            }
+        }
+
+        this.startDate = startDate;
+        this.startTime = startTime;
+        this.endDate = endDate;
+        this.endTime = endTime;
+    }
+
+    /**
+     * Prints the event's details, including its status, name, and start and end dates and times.
+     * If the start and end dates and times are not null, they are formatted and printed; otherwise,
+     * the raw input strings are printed.
+     */
+    @Override
+    public void printActivity() {
+        if (startDate != null && startTime != null && endDate != null && endTime != null) {
```

Comment:
null check is not recommended could cause problems in swe
https://stackoverflow.com/questions/1274792/is-returning-null-bad-design